be/src/storage/lake/transactions.cpp

```diff
@@ -109,7 +165,8 @@ StatusOr<TabletMetadataPtr> publish_version(TabletManager* tablet_mgr, int64_t t
     // 5. txn4 will be published in later publish task, but we can't judge what's the latest_version in BE and we can not reapply txn_log if
     // txn logs have been deleted.
     bool delete_txn_log = (txn_ids.size() == 1);
-    for (int i = 0; i < txn_ids.size(); i++) {
+    int txn_offset = ori_base_version < base_version ? base_version - ori_base_version : 0;
+    for (size_t i = txn_offset; i < txn_ids.size(); i++) {
```

Comment:
Should also update the if condition at lline 175?

Reply:
@smartlxh PTAL

Reply:
The txn log should not be not found after `adjust_base_version` and an error should be reported directly